### ISSUE TITLE
Add support for images in the listing description

### ISSUE THREAD

**Issue by a20hek** [SuperteamDAO/earn#478]
This issue is a paid bounty live on Superteam Earn right now, with a prize pool of $125 USDC. Check it out [here](https://earn.superteam.fun/listings/bounty/st-earn-oss-issue-image-support-in-the-listing-description/).

## Expected Behavior
Click on the upload image icon in the listing editor toolbox, and select the image (png or jpeg) to be uploaded to the listing description.

Uploaded images should be compressed before being added to our Cloudinary to save on server costs.  

Things to note: 
1. Images should always be centre-aligned 
2. Resizing the image should be possible but should not break the UI
3. Only jpeg or png formats should be supported 
4. Place the [upload icon](https://imgur.com/a/8rjZANY) between the Underline and Embed Link icons in the listing editor
<img width="101" alt="image" src="https://github.com/SuperteamDAO/earn/assets/67309069/1fe1e1c0-6f20-41b6-b26d-5899d6265b18">
<img width="929" alt="image" src="https://github.com/SuperteamDAO/earn/assets/67309069/c611937c-5319-4dfc-88d8-5704e3f16320">

## Actual Behavior
Users need to upload their images to a third-party website and link the uploaded image to the listing descriptions, leading to suboptimal UX for both talent and sponsors.

**Comment by apneduniya** [comment]
I would like to work on this

**Comment by a20hek** [comment]
**everyone can start working simultaneously**. since this is a bounty, I won't be assigning this issue to anyone
will be treated on a FCFS basis. **The best implementation will get merged**

**Comment by a20hek** [comment]
can take it up @Bot-Rakshit

**Comment by PratyushSawan** [comment]
Hi @a20hek While I was working on this issue, I figured out that there was a break in UI when I was adding the upload icon b/w Underline and Embed Link icons. Here you can see in the SS:
![earn_opensource](https://github.com/SuperteamDAO/earn/assets/40737575/4638c97c-3508-4d55-a09e-4a59d1082519)

I figured out a way to resolve this issue in my end. We can add our all Heading options (H1, H2, H3.. etc) in a select dropdown. So We can have proper spacing for all of our items and it's a clutter-free way to arrange the icons. Here is a small demo of my implementation Please have a look at this and give you feedback. I am still working on this to add an upload feature Thankyou.

![New UI feature update demo](https://github.com/SuperteamDAO/earn/assets/40737575/ae10c369-4298-49ed-afaa-597b67b87efe)

**Comment by Sajal0208** [comment]
Are you guys working on this issue? @PratyushSawan @Bot-Rakshit 
If yes, how much have you guys done? 

I was thinking of starting it!

**Comment by Sajal0208** [comment]
on it!

**Comment by dew-dr0p** [comment]
would also love to jump on

**Comment by a20hek** [comment]
try to reduce the size of the other buttons to make space for the image upload button?

**Comment by PratyushSawan** [comment]
> try to reduce the size of the other buttons to make space for the image upload button?

Hi @a20hek, I appreciate you suggestion! I did explore the option of adding more buttons to the compact toolbar, but I realize that it might become restrictive in the future if we need to accommodate additional options.

As as solution, I've implemented a dropdown menu for selecting heading. This approach not only maintains the compactness of the toolbar but also provides scalability for future additions. It's a more versatile solution that allows us to include more options without compromising on space or usability.

I'd love to hear you thoughts on this implementation and any suggestions you might have for further improvement.
